src/p_spec.c

```diff
@@ -2721,61 +2715,79 @@ void P_FreeScrollers (void)
 
 void R_InterpolateTextureOffsets (void)
 {
-  if (uncapped && leveltime > oldleveltime && !frozen_mode)
+  int i;
+
+  for (i = 0; i < numscrollers; i++)
   {
-    int i;
+    scroll_t *s = scrollers[i];
+    int dx, dy;
 
-    for (i = 0; i < numscrollers; i++)
+    if (s->accel)
     {
-      scroll_t *s = scrollers[i];
-      int dx, dy;
-
-      if (s->accel)
-      {
-        dx = s->vdx;
-        dy = s->vdy;
-      }
-      else
-      {
-        dx = s->dx;
-        dy = s->dy;
-
-        if (s->control != -1)
-        {   // compute scroll amounts based on a sector's height changes
-          fixed_t height = sectors[s->control].floorheight +
-            sectors[s->control].ceilingheight;
-          fixed_t delta = height - s->last_height;
-          dx = FixedMul(dx, delta);
-          dy = FixedMul(dy, delta);
-        }
+      dx = s->vdx;
+      dy = s->vdy;
+    }
+    else
+    {
+      dx = s->dx;
+      dy = s->dy;
+
+      if (s->control != -1)
+      {   // compute scroll amounts based on a sector's height changes
+        fixed_t height = sectors[s->control].floorheight +
+          sectors[s->control].ceilingheight;
+        fixed_t delta = height - s->last_height;
+        dx = FixedMul(dx, delta);
+        dy = FixedMul(dy, delta);
       }
+    }
 
-      if (!dx && !dy)
-        continue;
+    if (!dx && !dy)
+      continue;
 
-      switch(s->type)
-      {
-        side_t *side;
-        sector_t *sec;
+    switch(s->type)
+    {
+      side_t *side;
+      sector_t *sec;
 
-        case sc_side:
-          side = sides + s->affectee;
+      case sc_side:
+        side = sides + s->affectee;
+        if (uncapped && leveltime > oldleveltime && side->textureoffset != side->basetextureoffset)
```

Comment:
If we go with this code, could you factor out the common part of the checks into a new variable or macro?

Reply:
The problem with this check is that there could be two reasons why `side->textureoffset != side->basetextureoffset`: First, this could be the second rendered frame after the last calculated game tic. Or, this could be the second scroller affecting the same texture during the same rendered frame. In the first case, the delta value must get applied to the base offset. In the second case, the delta values must get accumulated. However, we currently have no means to distinguish both cases from each other.

Reply:
So should we add another field to `side_t` and other structures to distinguish between these cases?

Reply:
Maybe. This could be a solution. 🤷

Reply:
I'm afraid we need two loops then, one to accumulate the deltas and another one to actually apply them.

Reply:
This https://github.com/fabiangreffrath/woof/pull/1219/commits/3c19e60b77ac4f48cfc6e8648cd2ac12429ed3dd fixes all issues for me. Although it could probably be simplified somehow.